src/Microsoft.AspNetCore.WebUtilities/HttpRequestStreamReader.cs

```diff
@@ -132,9 +110,9 @@ protected override void Dispose(bool disposing)
 
         public override int Peek()
         {
-            if (_stream == null)
+            if (_disposed)
             {
-                throw new ObjectDisposedException("stream");
+                throw new ObjectDisposedException(nameof(HttpRequestStreamReader));
```

Comment:
Eh, that's fine. I looked through the CoreFX repo and in there it's a mix of `null`, `GetType().Name`, and other stuff like that. So `nameof(this_class_name)` seems fine.